src/tiled/mapdocumentactionhandler.cpp

```diff
@@ -299,16 +302,64 @@ void MapDocumentActionHandler::selectAll()
     }
 }
 
+void MapDocumentActionHandler::selectInverse()
+{
+    if (!mMapDocument)
+        return;
+
+    Layer *layer = mMapDocument->currentLayer();
+    if (!layer)
+        return;
+
+    if (TileLayer *tileLayer = layer->asTileLayer()) {
+        QRegion all(tileLayer->x(), tileLayer->y(),
+                  tileLayer->width(), tileLayer->height());
+
+        /*if (mMapDocument->selectedArea().isEmpty()) {
+            selectAll();
+        } else if(mMapDocument->selectedArea() == all) {
+            selectNone();
+        } else {*/
+            QUndoCommand *command = new ChangeSelectedArea(mMapDocument, all - mMapDocument->selectedArea());
+            mMapDocument->undoStack()->push(command);
+        /*}*/
+    } else if (ObjectGroup *objectGroup = layer->asObjectGroup()) {
+        QList<MapObject*> allObjects = objectGroup->objects();
+        QList<MapObject*> selectedObjects = mMapDocument->selectedObjects();
```

Comment:
No, I meant:
```c++
const QList<MapObject*> &allObjects = objectGroup->objects();
```
though you may also write:
const auto &allObjects = objectGroup->objects();